Tests/FluentAssertions.Specs/Xml/XElementAssertionSpecs.cs

```diff
@@ -1228,5 +1228,185 @@ public void When_asserting_element_has_a_child_element_with_an_empty_name_it_sho
         }
 
         #endregion
+
+        #region BeSingle
+
+        [Fact]
+        public void When_asserting_document_has_a_single_child_element_and_it_does_it_should_succeed()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                  </parent>");
+
+            // Act / Assert
+            document.Should().HaveElement("child").Which.Should().BeSingle();
+        }
+
+        [Fact]
+        public void When_asserting_document_has_single_child_element_but_it_does_have_two_it_should_fail()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                    <child />
+                  </parent>");
+
+            // Act
+            Action act = () => document.Should().HaveElement("child").Which.Should().BeSingle();
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage(
+                "Expected child to have a count of 1, but found 2.");
+        }
+
+        [Fact]
+        public void When_asserting_xElement_is_a_single_child_element_it_should_succeed()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                  </parent>");
+
+            // ReSharper disable once PossibleNullReferenceException
+            // Root is never null
+            var xElement = document.Root.Element("child");
+
+            // Act / Assert
+            xElement.Should().BeSingle();
```

Comment:
For me, such assertion does not make sense: An `XElement` is always a single element.

Reply:
I tested it in case the user gets one element from "somewhere" and wants also to check if it was a single element in his parent document. But now with xDocument as starting point for assertion, it's fixed.